Packs/VersaDirector/Integrations/VersaDirector/VersaDirector.py

```diff
@@ -0,0 +1,3546 @@
+
+        return response
+
+    def appliances_group_list_by_organization_request(self, organization: str = None, offset: int = None, limit: int = None):
+        params = assign_params(organization_name=organization, offset=offset, limit=limit)
+        headers = self._headers
+
+        response = self._http_request(
+            "GET",
+            url_suffix="nextgen/deviceGroup",
+            params=params,
+            headers=headers,
+        )
+
+        return response
+
+    def appliances_list_by_device_group_request(
+        self, device_group: str = None, template_name: str = None, offset: int = None, limit: int = None
+    ):
+        params = assign_params(
+            offset=offset,
+            limit=limit,
+        )
+
+        response = self._http_request(
+            "GET",
+            url_suffix=f"nextgen/deviceGroup/{device_group}/template/{template_name}",
+            params=params,
+            headers=self._headers,
+        )
+
+        return response
+
+    def template_list_by_organization_request(self, organization: str = None, type: list = None, offset: int = None, limit: int = None):
+        params = assign_params(organization=organization, type=type, offset=offset, limit=limit)
+        headers = self._headers
+
+        response = self._http_request("GET", url_suffix="vnms/template/metadata", params=params, headers=headers)
+
+        return response
+
+    def template_list_by_datastore_request(self, organization: str = None, offset: int = None, limit: int = None):
+        params = assign_params(offset=offset, limit=limit)
+        headers = self._headers
+        headers["Content-Type"] = "application/xml"
+        response = self._http_request(
+            "GET",
+            url_suffix=f"api/config/devices/template/{organization}-DataStore/config/orgs/org",
+            params=params,
+            headers=headers,
+            resp_type="xml",
+        )
+
+        return response
+
+    def application_service_template_list_request(
+        self, organization: str = None, keyword: str = None, offset: int = None, limit: int = None
+    ):
+        params = assign_params(organization=organization, searchKeyword=keyword, offset=offset, limit=limit)
+        response = self._http_request("GET", url_suffix="/nextgen/applicationServiceTemplate", params=params, headers=self._headers)
+
+        return response
+
+    def template_change_commit_request(self, template_name: str, appliances: list, mode: str, reboot: str):
+        request_body = {"versanms.templateRequest": {"device-list": appliances, "mode": f"{mode}", "reboot": f"{reboot}"}}
+
+        response = self._http_request(
+            "POST",
+            url_suffix=f"vnms/template/applyTemplate/{template_name}/devices",
+            headers=self._headers,
+            json_data=request_body,
+            ok_codes=(200,),
```

Comment:
As previously stated, ok_codes default value is insufficient because this http request should only accept status code 200.
I will add comments to the code to clarify this.